### ISSUE TITLE
Request implementation of SSE regularization

### ISSUE THREAD

**Issue by wuliwei9278** [tensorflow/tensorflow#45030]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.3 (also used 1.14 before)
- Are you willing to contribute it (Yes/No): 
I would like to help but I am not too familiar with tensorflow operator level coding.

**Describe the feature and the current behavior/state.**
SSE (stochastic shared embedding) has been proven to be a very useful embedding level regularization techniques in deep learning when dropout and weight decay do not work well enough for embeddings. 

See paper: "Stochastic Shared Embeddings: Data-driven Regularization of Embedding Layers": https://papers.nips.cc/paper/2019/file/37693cfc748049e45d87b8c7d8b9aacd-Paper.pdf; "SSE-PT: Sequential Recommendation Via Personalized Transformer"(https://dl.acm.org/doi/10.1145/3383313.3412258); "Multimodal Categorization of Crisis Events in Social Media" (https://openaccess.thecvf.com/content_CVPR_2020/papers/Abavisani_Multimodal_Categorization_of_Crisis_Events_in_Social_Media_CVPR_2020_paper.pdf). 

But this feature is largely missing in tensorflow implementation and makes it hard for users to use SSE, unlike dropout. Ideally it should be as easy to use SSE as using dropout or l2.

**Will this change the current api? How?**
Yes, this would require a new embedding_lookup operator that supports embeddings stochastic transitioning according to a predefined sparse probability transition matrix or a uniform transition mechanism at default. The uniform case is implemented at https://github.com/wuliwei9278/SSE-PT as a simple example but this is largely a hack. 

**Who will benefit with this feature?**
I work at Linkedin and when training embeddings using this regularization technique in addition to dropout and l2 would lead to faster convergence of neural nets and better performance but I had to use it using a hack by duplicating training data with embedding ids masked into different ids, which is not very efficient and cannot be scaled when dataset size is large. Including this would make my life much easier, and likewise for my colleagues and many researchers, whose models also suffer from model embedding layer overfitting.

**Any Other info.**
Please contact liwwu@linkedin.com for more information.

**Comment by wuliwei9278** [comment]
Just want to follow up to see if there are any questions or concerns that I can help to address. Happy Thanksgiving!

**Comment by wuliwei9278** [comment]
any progress on this?

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
